subsys/net/ip/tcp.c

```diff
@@ -2253,6 +2255,15 @@ int net_tcp_queue_data(struct net_context *context, struct net_pkt *pkt)
 
 	len = net_pkt_get_len(pkt);
 
+	if (IS_ENABLED(CONFIG_NET_CONTEXT_SNDBUF) &&
+		net_context_get_option(context, NET_OPT_SNDBUF, &sndbuf, &sndbuf_len) == 0) {
+		if ((sndbuf > 0) &&
+			(net_pkt_get_len(conn->send_data) + len > sndbuf)) {
+			ret = -ENOMEM;
+			goto out;
+		}
+	}
+
```

Comment:
I don't think we should use the option like this. First, it duplicates the logic of TX window handling, second, it will end up in ENOMEM socket errors if we attempt to send too much at once.

I think it'd be better to modify the logic that determines the maximum TX window size [here](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/ip/tcp.c#L1790). In case `NET_OPT_SNDBUF` is set on a context, use the option value for `max_win`, otherwise use the current, Kconfig-based defaults. The existing logic will make sure that the window is not exceeded.